src/serac/physics/materials/thermal_material_input.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019-2023, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_material_input.hpp
+ *
+ * @brief This file contains functions for reading a material from input files
+ */
+
+#pragma once
+
+#include <string>
+#include "serac/infrastructure/input.hpp"
+#include "serac/physics/materials/thermal_material.hpp"
+
+namespace serac {
+
+/// @brief Holds all possible heat transfer materials that can be utilized in our Input Deck
+using var_thermal_material_t = std::variant<heat_transfer::LinearIsotropicConductor, heat_transfer::LinearConductor<2>,
+                                            heat_transfer::LinearConductor<3>>;
```

Comment:
This is a reasonable implementation for what we currently offer, but this design isn't extensible for users that want to add their own material models.

I think we may want to reconsider this approach so that users can more easily write new materials and access them from input files without modifying serac. For example (although not necessarily a good implementation), instead of a `std::variant`, we could have a user-extendable `std::map<std::string, std::function< ... > >` of callbacks. After reading the name of the material model, it would invoke the callback in that std::map (where the user could write code to set up their custom material model, and then register it to the physics module with `setMaterial`).

Reply:
@white238 , we probably want this in the future. Is it best to leave it at the current implementation until we add this capability or would the `std::map` approach be better?

Reply:
sorry for not clarifying in the initial comment: I'm not expecting any changes in this PR-- just that we might want to discuss and revisit this design choice at some point in the future